src/terminal/parser/stateMachine.cpp

```diff
@@ -1292,10 +1681,12 @@ void StateMachine::ProcessCharacter(const wchar_t wch)
         _ActionExecute(wch);
         _EnterGround();
     }
-    else if (_isEscape(wch) && _state != VTStates::OscString)
+    else if (_isEscape(wch) && _state != VTStates::OscString && _state != VTStates::DcsPassThrough)

```

Comment:
do we need a helper, `_canStateBeTerminatedWithEscape()`?

Reply:
That will be in my following PR where the code structure is changed.